### PR TITLE
Image zZoom für Alex

### PR DESCRIPTION
**Pull request by pejotejo**
## Why? What?

Make image panel zoomable. This makes the image panel function more consistently to other panels. Really helpful for calibrating pose detection, as the confidences around the pose keypoints are really close together and otherwise not distinguishable.  

## How to Test

- scroll and drag the image in the image panel

### CODE REVIEW

File: tools/twix/src/panels/image/mod.rs [hunk 1]
```diff
@@ -151,36 +153,49 @@ impl Widget for &mut ImagePanel {
...
-        match self.show_image(ui) {
-            Ok(response) => response,
-            Err(error) => ui.label(format!("{error:#?}")),
-        }
+        let (rect, response) = ui.allocate_at_least(vec2(640.0, 480.0), Sense::click_and_drag());
+        let mut painter = TwixPainter::paint_at(ui, rect).with_camera(
+            vector![640.0, 480.0],
+            Similarity2::identity(),
+            CoordinateSystem::LeftHand,
+        );
```
Comment (ThagonDuarte) [inline tools/twix/src/panels/image/mod.rs:161]: While the image panel can now use zoom and pan operations, the resizing of the image when resizing the twix panel somehow broke.

Was there a particular reason, you allocated the `TwixPainter` the way that you did, and not like for example how it is done in the map panel (from which I have taken the suggestion)? The suggestion below fixed he resizing problem and still allows for zooming and panning. 
```suggestion
        let (response, painter) = TwixPainter::allocate_new(ui);
        let mut painter = painter.with_camera(
            vector![640.0, 480.0],
            Similarity2::identity(),
            CoordinateSystem::LeftHand,
        );
```

**Review by ThagonDuarte — CHANGES_REQUESTED** [review]
Overall looks good. Just the bug with the image not resizing to the panel size should be changed.

**Review by ThagonDuarte — CHANGES_REQUESTED** [review]
Oh and the twix version should be bumped.
In `tools/twix/Cargo.toml`: Increment package version.

File: tools/twix/src/panels/image/mod.rs (hunk 1 above)
Comment (pejotejo) [reply tools/twix/src/panels/image/mod.rs]: done

**Review by oleflb — CHANGES_REQUESTED** [review]

File: tools/twix/src/twix_painter.rs [hunk 2]
```diff
@@ -113,6 +113,14 @@ impl<Frame> TwixPainter<Frame> {
         self.world_to_pixel = transformation * self.world_to_pixel;
     }
 
+    pub fn context(&self) -> &Context {
+        self.painter.ctx()
+    }
+
+    pub fn is_right_handed(&self) -> bool {
+        self.camera_coordinate_system.y_scale() >= 0.0
```
[the remark this answers is not in the document]
Comment (pejotejo) [reply tools/twix/src/twix_painter.rs]: done

**Comment by pejotejo** [comment]
@ThagonDuarte what is the state of the review?

File: tools/twix/src/main.rs [hunk 3]
```diff
@@ -59,6 +59,7 @@ mod selectable_panel_macro;
 mod twix_painter;
 mod value_buffer;
 mod visuals;
+pub mod zoom_and_pan;
```
Comment (oleflb) [inline tools/twix/src/main.rs:62]: remove `pub` if not required

File: tools/twix/src/zoom_and_pan.rs [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+use nalgebra::{vector, Similarity2, Translation2};
+
+use crate::twix_painter::TwixPainter;
+
+#[derive(Default)]
+pub struct ZoomAndPanManager {
+    transformation: Similarity2<f32>,
+}
+
+impl ZoomAndPanManager {
```
Comment (oleflb) [inline tools/twix/src/zoom_and_pan.rs:12]: hmm what do you think of `ZoomAndPanTransform` instead?

File: tools/twix/src/zoom_and_pan.rs [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+        painter.append_transform(zoom_transform.framed_transform());
+        let pointer_in_pixel_after_zoom =
+            painter.transform_world_to_pixel(pointer_in_world_before_zoom);
+        let shift_from_zoom = pointer_position - pointer_in_pixel_after_zoom;
+        let pixel_drag = vector![response.drag_delta().x, response.drag_delta().y];
+        self.transformation.append_scaling_mut(zoom_factor);
+        let zoom_shift = vector![shift_from_zoom.x, shift_from_zoom.y];
+        self.transformation
+            .append_translation_mut(&Translation2::from(pixel_drag + zoom_shift));
+    }
```
Comment (oleflb) [inline tools/twix/src/zoom_and_pan.rs:39]: please insert newline between functions

File: tools/twix/src/zoom_and_pan.rs [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+    pub fn transformation(&self) -> Similarity2<f32> {
+        self.transformation
```
Comment (oleflb) [inline tools/twix/src/zoom_and_pan.rs:41]: what do you think of making `self.transformation` `pub` instead?

File: tools/twix/src/twix_painter.rs [hunk 7]
```diff
@@ -4,55 +4,48 @@ use std::{
...
+type ScreenTransform<Frame> = Transform<Frame, Screen, Affine2<f32>>;
 pub struct TwixPainter<Frame> {
     painter: Painter,
     pixel_rect: Rect,
-    world_to_pixel: Similarity2<f32>,
-    camera_coordinate_system: CoordinateSystem,
+    world_to_pixel: ScreenTransform<Frame>,
     frame: PhantomData<Frame>,
 }
+trait Convert {
```
Comment (oleflb) [inline tools/twix/src/twix_painter.rs:25]: please insert newlines between struct and trait definitions

File: tools/twix/src/twix_painter.rs [hunk 8]
```diff
@@ -91,28 +83,41 @@ impl<Frame> TwixPainter<Frame> {
...
+        &mut self,
+        transformation: Transform<Screen, Screen, Similarity2<f32>>,
+    ) {
         self.world_to_pixel = transformation * self.world_to_pixel;
     }
 
+    pub fn context(&self) -> &Context {
+        self.painter.ctx()
+    }
+    pub fn scaling(&self) -> f32 {
```
Comment (oleflb) [inline tools/twix/src/twix_painter.rs:112]: please insert newline

**Review by oleflb — CHANGES_REQUESTED** [review]

File: tools/twix/src/zoom_and_pan.rs (hunk 6 above)
Comment (pejotejo) [reply tools/twix/src/zoom_and_pan.rs]: done

File: tools/twix/src/zoom_and_pan.rs (hunk 5 above)
Comment (pejotejo) [reply tools/twix/src/zoom_and_pan.rs:39]: removed function

File: tools/twix/src/zoom_and_pan.rs (hunk 4 above)
Comment (pejotejo) [reply tools/twix/src/zoom_and_pan.rs]: sounds good, done

File: tools/twix/src/main.rs (hunk 3 above)
Comment (pejotejo) [reply tools/twix/src/main.rs]: done

File: tools/twix/src/twix_painter.rs (hunk 8 above)
Comment (pejotejo) [reply tools/twix/src/twix_painter.rs:115]: done

File: tools/twix/src/twix_painter.rs (hunk 7 above)
Comment (pejotejo) [reply tools/twix/src/twix_painter.rs:26]: done

**Review by oleflb — APPROVED** [review]

**Review by oleflb — APPROVED** [review]

**Review by ThagonDuarte — APPROVED** [review]
LGTM!

**Pull request merged**